src/components/LoadingPurchase.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { ContractHelper } from "../services/contract";
+import Popup from "./Popup";
+import RevealComplete from "./RevealComplete";
+
+type Props = {
+  show: boolean;
+  close: () => void;
+  loading: boolean;
+};
+
+const LoadingPurchase: React.FC<Props> = (props) => {
+  const [loading, setLoading] = useState(false);
+  const [revealed, setRevealed] = useState(false);
+
+  const reveal = async () => {
+    const contractHelper = new ContractHelper();
+    await contractHelper.init();
+    const dungeonContract = await contractHelper.getDungeonContract();
+    dungeonContract.methods
+      .revealWaifus()
+      .send()
+      .on("transactionHash", (hash: any) => {
+        setLoading(true);
+      })
+      .on("receipt", (receipt: any) => {
+        setLoading(false);
+        setRevealed(true);
+      })
+      .on("error", (err: any) => {
+        alert(`Error: ${err.message}`);
+      });
+  };
+
+  return (
```

Comment:
We need a better loading indicator here, but that can prob be implemented once the other loading PR is merged

Reply:
Have added the loading spinner 😄